packages/node_modules/pouchdb-find/src/adapters/local/find/index.js

```diff
@@ -89,119 +87,113 @@ function find(db, requestDef, explain) {
-    }, queryPlan.queryOpts);
+  var opts = Object.assign({
+    include_docs: true,
+    reduce: false,
+    // Add amount of index for doAllDocs to use (related to issue #7810)
+    indexes_count: getIndexesRes.total_rows,
+  }, queryPlan.queryOpts);
 
-    if ('startkey' in opts && 'endkey' in opts &&
-        collate(opts.startkey, opts.endkey) > 0) {
-      // can't possibly return any results, startkey > endkey
-      /* istanbul ignore next */
-      return {docs: []};
-    }
+  if ('startkey' in opts && 'endkey' in opts &&
+      collate(opts.startkey, opts.endkey) > 0) {
+    // can't possibly return any results, startkey > endkey
+    /* istanbul ignore next */
+    return {docs: []};
+  }
 
-    var isDescending = requestDef.sort &&
-      typeof requestDef.sort[0] !== 'string' &&
-      getValue(requestDef.sort[0]) === 'desc';
+  var isDescending = requestDef.sort &&
+    typeof requestDef.sort[0] !== 'string' &&
+    getValue(requestDef.sort[0]) === 'desc';
 
-    if (isDescending) {
-      // either all descending or all ascending
-      opts.descending = true;
-      opts = reverseOptions(opts);
-    }
+  if (isDescending) {
+    // either all descending or all ascending
+    opts.descending = true;
+    opts = reverseOptions(opts);
+  }
 
-    if (!queryPlan.inMemoryFields.length) {
-      // no in-memory filtering necessary, so we can let the
-      // database do the limit/skip for us
-      if ('limit' in requestDef) {
-        opts.limit = requestDef.limit;
-      }
-      if ('skip' in requestDef) {
-        opts.skip = requestDef.skip;
-      }
+  if (!queryPlan.inMemoryFields.length) {
+    // no in-memory filtering necessary, so we can let the
+    // database do the limit/skip for us
+    if ('limit' in requestDef) {
+      opts.limit = requestDef.limit;
     }
-
-    if (explain) {
-      return Promise.resolve(queryPlan, opts);
+    if ('skip' in requestDef) {
+      opts.skip = requestDef.skip;
     }
+  }
 
-    return Promise.resolve().then(function () {
-      if (indexToUse.name === '_all_docs') {
-        return doAllDocs(db, opts);
-      } else {
-        var signature = indexToSignature(indexToUse);
-        return abstractMapper(db).query.call(db, signature, opts);
-      }
-    }).then(function (res) {
-      if (opts.inclusive_start === false) {
-        // may have to manually filter the first one,
-        // since couchdb has no true inclusive_start option
-        res.rows = filterInclusiveStart(res.rows, opts.startkey, indexToUse);
-      }
+  if (explain) {
+    return Promise.resolve(queryPlan, opts);
+  }
 
-      if (queryPlan.inMemoryFields.length) {
-        // need to filter some stuff in-memory
-        res.rows = filterInMemoryFields(res.rows, requestDef, queryPlan.inMemoryFields);
-      }
+  const res = indexToUse.name === '_all_docs'
+    ? await doAllDocs(db, opts)
+    : await abstractMapper(db).query.call(db, indexToSignature(indexToUse), opts);
```

Comment:
I find this hard to follow. I think it's multiline ternary statements with awaits. A cleaner solution for me is to pull out a named function to do this, eg:

```js
async function queryIndex(db, opts, indexToUse) { // TODO come up with a better fn name
  if (indexToUse.name === '_all_docs') {
    return doAllDocs(db, opts);
  }
  return abstractMapper(db).query.call(db, indexToSignature(indexToUse), opts);
}

...

const res = await queryIndex(db, opts, indexToUse);
```

Reply:
https://github.com/pouchdb/pouchdb/pull/8914/commits/1edb94a6224e82c01f049eb6d708f67e9f127655 should do the trick ;)